utilities/ovs-ofctl.c

```diff
@@ -4182,6 +4237,7 @@ static const struct ovs_cmdl_command all_commands[] = {
     { "parse-key-value", NULL, 1, INT_MAX, ofctl_parse_key_value },
 
     { "load-filter-prog", "id file [length]", 2, 3, ofctl_load_filter_prog },
+    { "update-map", "prog_id map_id key value", 3, INT_MAX, ofctl_update_map },
```

Comment:
Isn't `min_arg` actually 5 here (instead of 3)? `update-map [prog_id] [map_id] key [key] value [value]`.

Reply:
I'm not sure how does the OVS treat arguments if the format is: "ovs-ofctl update-map br0 1 0 key 1 0 0 0 value 1 0 ..."? It seems that in case of min_arg=5 the first byte of key would be treated as one of arguments? It's hard to predict the number of bytes/length of key and value.

Reply:
As far as I can see `min_args` is only used to [ensure the command has at least that number of arguments](https://github.com/Orange-OpenSource/oko/blob/11208312300f78a015a1f3d13762d01b38ca5507/lib/command-line.c#L114-L116), not for the actual parsing logic. Am I missing something?

Reply:
That's correct. What I mean is that the `value` arg will not be taken into account during `n_arg` checking as it is preceded with bytes of key. So, in case of `min_arg` = 5 the arguments that will be checked against `min_arg` would be: `[bridge_id] [prog_id] [map_id] key` plus the first byte of key's content. So, effectively the `min_arg` should be set to 4 to avoid taking the first byte into account. However, I've assumed that once checking for the existence of `key` and `value` is done during argument parsing I can check here only those three arguments without `key` and `value`. 

Maybe, I'm missing something...

Reply:
> So, effectively the `min_arg` should be set to 4 to avoid taking the first byte into account.

Why shouldn't the first byte be taken into account? We don't want to allow keys (respectively, values) of zero length, right?

My `min_args=5` was incorrect actually, I meant 6 since we'd have 6 arguments minimum with `update-map [prog_id] [map_id] key [one-byte key] value [one-byte value]` (I don't think the bridge name is counted, see `min_args` for `load-filter-prog`). That would be the shortest `update-map` command we can write.

Reply:
Now, it makes sense. Regarding bridge name I need to double-check.

Reply:
Hm. Looks like you're right and `load-filter-prog`'s `min_args` and message are incorrect. I'll need to fix that.